src/blob_gc_job.cc

```diff
@@ -477,15 +478,22 @@ Status BlobGCJob::InstallOutputBlobFiles() {
         tmp.append(" ");
       }
       tmp.append(std::to_string(file->file_number()));
-
-      blob_gc_->AddOutputFile(file.get());
+      // To temporarily hold the ptr of BlobFileMeta
+      // FilesMeta will be added to blob_gc after successful
+      // BatchFinishFiles
+      tmp_file_metas.push_back(file);
       files.emplace_back(std::make_pair(file, std::move(builder.first)));
     }
     ROCKS_LOG_BUFFER(log_buffer_, "[%s] output[%s]",
                      blob_gc_->column_family_handle()->GetName().c_str(),
                      tmp.c_str());
-    this->blob_file_manager_->BatchFinishFiles(
+    s = this->blob_file_manager_->BatchFinishFiles(
         blob_gc_->column_family_handle()->GetID(), files);
+    if (s.ok()) {
+      for (auto file : tmp_file_metas) {
```

Comment:
why not use file meta in `files`?

Reply:
this will cause compile error...I have not figured out the reason

Reply:
I figured out it, because I use `for(auto file : files)`, but files is consisted of pair<share_ptr, unique_ptr> in which unique_ptr cannot be copied, correct way is `for(auto &file : files)`